src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/innerassignment/InputInnerAssignment.java

```diff
@@ -128,5 +128,50 @@ void method() throws IOException {
         while (line < 3 || (line = file.read()) != null) {}
         while (line < 3 & (line = file.read()) != null) {}
         while (line < 3 | (line = file.read()) != null) {}
+
+        do{}
+        while (!((line = file.read()) != null));
+        do{}
+        while ((line = file.read()) != null && line < 3);
```

Comment:
Just a question, this should be another issue if it is identified as one.

Will I atleast get a violation if I change `line < 3` to `line = 3` (assignment and not equals)?

Reply:
Changing `line < 3` to `line = 3`  will cause compiler error as it expects some boolean expression associated with `&& ` operator